client/plots/sampleScatter.renderer.js

```diff
@@ -104,7 +103,26 @@ export function setRenderers(self) {
 		if (chart.colorLegend.get('Ref')?.sampleCount > 0) colorLegendSize += 60
 		const scaleHeight = self.config.scaleDotTW ? 200 : 100
 		self.legendHeight = Math.max(colorLegendSize, chart.shapeLegend.size * 30) + scaleHeight //legend step and header
-		const width = self.charts.length == 1 ? s.svgw + 800 : s.svgw + (self.config.shapeTW ? 600 : 350)
+
+		//Dynamically calculate the length of the legend labels
+		const getLegendLabelWidth = (key, svg) => {
+			const legend = chart[`${key}Legend`]
+			if (!legend) return 0
+			const labels = []
+			for (const [k, v] of legend.entries()) {
+				if (k != 'Ref') labels.push(`${k}, n=(${v.sampleCount})`)
+			}
+			labels.push(self.config[`${key}TW`]?.term?.name ?? '')
+			const size = self.config[`${key}TW`]?.term?.type == 'geneVariant' ? 1.1 : 1
+			// Add 20 for the icon (16) and space
+			return getMaxLabelWidth(svg, labels, size) + 20
+		}
+		/** Becomes the x offset for the shape legend.
+		 * When in continuous mode, color scale renders with a
+		 * default width of 150. */
+		chart.colorLegendWidth = self.config?.colorTW?.q.mode == 'continuous' ? 175 : getLegendLabelWidth('color', svg)
```

Comment:
it should be` Math.max(175, getMaxLabelWidth(svg, [self.config?.colorTW?.term.name])) + 20 `
otherwise the term name may be clipped, like here 
<img width="273" alt="Screenshot 2025-01-24 at 4 54 30 PM" src="https://github.com/user-attachments/assets/a9a81d64-2977-4e0f-996c-67f082b58c6e" />